### DISCUSSION TITLE
Multiple git branches to the specific custom domain

### DISCUSSION THREAD

**Discussion by mununki** [vercel/vercel#8315] [Help] [answered]
Hi, team.
I've been using the custom domain configuration for the specific git branch, e.g. `staging`. But, I can not find how to assign the specific domain to the multiple git branches. For example, I'd like to assign the custom domain to all other git branches except `main`, `staging`. Is it achievable?

**Comment by amyegan** [comment, marked as answer]
Hi @mattdamon108. I'm not sure I understand what you're asking. A custom domain is assigned to a single branch (your production branch by default). You can assign additional domains or subdomains to different branches in your Project Settings. You can read more about it here: [Assigning a domain to a git branch](https://vercel.com/docs/concepts/projects/custom-domains#assigning-a-domain-to-a-git-branch)

**Reply by amyegan** [reply]
@mattdamon108 You can add subdomains [from the CLI ](https://vercel.com/docs/cli#commands/alias) or [from the API](https://vercel.com/docs/rest-api#endpoints/aliases). 

You should also take a look at our [Generated URL with Git Branch](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) doc. You can always get to the latest successful deployment from a git branch by visiting the URL for that branch. It will be formatted like `<project-name>-git-<branch-name>-<scope-slug>.vercel.app`, where scope-slug is your account/team name

**Reply by mununki** [reply]
I found out how to change the preview deployment suffix, but it seems to affect only the team, not the project. Is there any configuration for the project in GUI?

**Comment by mununki** [comment]
@amyegan Currently, I can assign the additional domain to each git branch. For example, I assigned the custom domain(`www.prod-domain.com`) to the main branch as production and an additional domain(`app-staging.prod-main.com`) to the staging branch. But I'd like to assign the additional domain for the rest of git branches to be pushed automatically. If I push the branch name `x`, then it is assigned to `x.additional-domain.dev`, I push branch name `y` -> `y.additional-domain.dev`. Does it make sense to you?
